app/api/routes/images.py

```diff
@@ -0,0 +1,137 @@
+"""
+Image upload and validation endpoints.
+"""
+
+import logging
+import uuid
+
+from datetime import UTC, datetime
+
+from fastapi import APIRouter, File, HTTPException, UploadFile
+from pydantic import BaseModel
+
+from app.services.image_validation import validate_image
+
+
+logger = logging.getLogger(__name__)
+router = APIRouter(prefix='/images', tags=['images'])
+
+
+class ImageUploadResponse(BaseModel):
+    """Response model for successful image upload."""
+
+    image_id: str
+    filename: str
+    content_type: str
+    size_bytes: int
+    width: int
+    height: int
+    uploaded_at: str
+    success: bool = True
+
+
+class ImageValidationError(BaseModel):
+    """Response model for image validation errors."""
+
+    error: str
+    detail: str
+    success: bool = False
+
+
+@router.post('/upload', response_model=ImageUploadResponse)
+async def upload_image(file: UploadFile = File(...)):
+    """
+    Upload and validate an image file.
+
+    This endpoint validates the uploaded image for:
+    - File type (jpg, jpeg, png, webp)
+    - File size (max 5 MB)
+    - Resolution (max 4096×4096 px)
+
+    Args:
+        file: The uploaded image file
+
+    Returns:
+        ImageUploadResponse with image metadata
+
+    Raises:
+        HTTPException: If validation fails (400) or processing error occurs (500)
+
+    """
+    try:
+        logger.info(f'Received image upload request: {file.filename}')
+
+        # Validate the image using the validation service
+        validation_result = await validate_image(file)
```

Comment:
validate_image(file) is awaited, but file pointer may not be reset before next use. Ensure file pointer is reset after calling validate_image.
```
validation_result = await validate_image(file)
await file.seek(0)
```